engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -141,9 +157,12 @@ class DeleteWorkflowFilesActor(rootWorkflowId: RootWorkflowId,
   }
 
   whenUnhandled {
+    case Event(ioReceivable, _) if ioReceive.isDefinedAt(ioReceivable) =>
```

Comment:
TOL how about just this one and not all the others?

Reply:
So even for messages which the other cases are catching (eg `case IoSuccess`) the client helpers sometimes want to have their chance first to do some bookkeeping (removing timeouts, etc) - so I think this is needed at the _top_ of every chain, rather than as a catch-all at the end.

Reply:
nm my original suggestion...

Isn't this pattern consuming the messages in `ioReceive` and not allowing `DeleteWorkflowFilesActor` to apply its own business logic?

Reply:
As far as I can tell, if the message it intercepted by `ioReceive` (eg because there's a pending timeout):
* The `ioReceive` will take the message and do its bookkeeping
* The `ioReceive` calls back into `receive` to handle the message
* This time, the `ioReceive` will no longer match the message, and it falls down to the actor-specific handler